blockchain/blockdao.go

```diff
@@ -455,29 +533,33 @@ func (dao *blockDAO) putBlock(blk *block.Block) error {
 
 	value, err = dao.kvstore.Get(blockNS, totalActionsKey)
 	if err != nil {
-		return errors.Wrap(err, "failed to get total actions")
+		err = errors.Wrap(err, "failed to get total actions")
+		return
 	}
 	totalActions := enc.MachineEndian.Uint64(value)
 	totalActions += uint64(len(blk.Actions))
 	totalActionsBytes := byteutil.Uint64ToBytes(totalActions)
 	batch.Put(blockNS, totalActionsKey, totalActionsBytes, "failed to put total actions")
 
 	if !dao.writeIndex {
-		return dao.kvstore.Commit(batch)
+		err = dao.kvstore.Commit(batch)
+		return
 	}
-	if err := indexBlock(dao.kvstore, blk, batch); err != nil {
-		return err
+	if err = indexBlock(dao.kvstore, blk, batch); err != nil {
+		return
 	}
-	return dao.kvstore.Commit(batch)
+	err = dao.kvstore.Commit(batch)
+	return
 }
 
 // putReceipts store receipt into db
-func (dao *blockDAO) putReceipts(blkHeight uint64, blkReceipts []*action.Receipt) error {
+func (dao *blockDAO) putReceipts(blkHeight uint64, blkReceipts []*action.Receipt, kvstore db.KVStore) error {
```

Comment:
Isn't an easier fix that preventing opening a new db file when putting receipt?

Reply:
need to think about it